### ISSUE TITLE
host reboot when i use curl to test bookinfo with Cilium 1.7.2 + sockops + istio 1.5.1 . 

### ISSUE THREAD

**Issue by seraphsea** [cilium/cilium#11106]
- Cilium version (run `cilium version`): 1.7.2
- Kernel version (run `uname -a`) 4.18.0-147.5.1.el8_1.x86_64(Centos 8.1)
- Orchestration system version in use (e.g. `kubectl version`, Mesos, ...)
 kubectl:1.14.2
docker-ce:18.09.1
- Link to relevant artifacts (policies, deployments scripts, ...)
- Upload a system dump (run `curl -sLO
https://github.com/cilium/cilium-sysdump/releases/latest/download/cilium-sysdump.zip &&
python cilium-sysdump.zip` and then attach the generated zip file)
**How to reproduce the issue**

there are only one host in this issue. 
use Self-Managed Kubernetes not minikube in this issue.

1. instruction 1
a) install Kubernetes 1.14.2 on Centos 8.1 
b) wget https://raw.githubusercontent.com/cilium/cilium/1.7.2/install/kubernetes/quick-install.yaml
c) add sockops-enable: "true" in the yaml file.
d) kubectl apply -f quick-install.yaml to install cilium 1.7.2
2. instruction 2
a) following the tutorial https://docs.cilium.io/en/stable/gettingstarted/istio/ to install istio 1.5.1
b) use curl to test if bookinfo works, when i setup bookinfo Application V1
c) host reboot and call trace as follow:

```
Apr 21 17:21:26 e69a07245.et15sqa kernel: WARNING: CPU: 16 PID: 78113 at lib/iov_iter.c:883 copy_page_to_iter+0xd9/0x310
Apr 21 17:21:26 e69a07245.et15sqa kernel: Modules linked in: xt_REDIRECT(E) xt_owner(E) xt_statistic(E) xt_TPROXY(E) nf_tproxy_ipv6(E) nf_tproxy_ipv4(E) xt_CT(E) vxlan(E) ip6_udp_tunnel(E) udp_tunnel(E) xt_socket(E) nf_socket_ipv4(E) nf_socket_ipv6(E) ip6table_filter(E) ip6table_raw(E) ip6table_mangle(E) ip6_tables(E) iptable_raw(E) iptable_mangle(E) xt_nat(E) ipt_REJECT(E) nf_reject_ipv4(E) ip_vs_sh(E) ip_vs_wrr(E) ip_vs_rr(E) ip_vs(E) xt_comment(E) xt_mark(E) xt_MASQUERADE(E) nf_conntrack_netlink(E) nfnetlink(E) iptable_nat(E) xt_addrtype(E) iptable_filter(E) xt_conntrack(E) nf_nat(E) nf_conntrack(E) nf_defrag_ipv6(E) nf_defrag_ipv4(E) libcrc32c(E) br_netfilter(E) binfmt_misc(E) overlay(E) veth(E) bridge(E) stp(E) llc(E) intel_rapl_msr(E) intel_rapl_common(E) iosf_mbi(E) x86_pkg_temp_thermal(E) coretemp(E) kvm_intel(E) kvm(E) bonding(E) irqbypass(E) crct10dif_pclmul(E) crc32_pclmul(E) ghash_clmulni_intel(E) iTCO_wdt(E) iTCO_vendor_support(E) mxm_wmi(E) aesni_intel(E) crypto_simd(E) cryptd(E) glue_helper(E) mei_me(E)
Apr 21 17:21:26 e69a07245.et15sqa kernel:  lpc_ich(E) sg(E) ioatdma(E) mei(E) pcspkr(E) mfd_core(E) i2c_i801(E) ipmi_si(E) ipmi_devintf(E) wmi(E) ipmi_msghandler(E) acpi_power_meter(E) ip_tables(E) ext4(E) mbcache(E) jbd2(E) sd_mod(E) ast(E) drm_vram_helper(E) ttm(E) crc32c_intel(E) nvme(E) nvme_core(E) drm_kms_helper(E) syscopyarea(E) sysfillrect(E) sysimgblt(E) fb_sys_fops(E) drm(E) ixgbe(E) igb(E) mdio(E) i2c_algo_bit(E) dca(E) i2c_core(E) ahci(E) libahci(E) libata(E)
Apr 21 17:21:26 e69a07245.et15sqa kernel: CPU: 16 PID: 78113 Comm: envoy Tainted: G        W   E     5.4.32 #1
Apr 21 17:21:26 e69a07245.et15sqa kernel: Hardware name: Inspur SA5212M4/YZMB-00370-109, BIOS 4.1.12 06/18/2017
Apr 21 17:21:26 e69a07245.et15sqa kernel: RIP: 0010:copy_page_to_iter+0xd9/0x310
Apr 21 17:21:26 e69a07245.et15sqa kernel: Code: 85 18 01 00 00 48 01 f2 48 39 d5 77 1d 48 8b 39 be 00 10 00 00 f7 c7 00 00 01 00 0f 85 df 00 00 00 48 39 f2 0f 86 52 ff ff ff <0f> 0b 31 ed eb b0 83 e2 fe 83 fa 40 74 a8 83 fa 20 0f 84 56 01 00
Apr 21 17:21:26 e69a07245.et15sqa kernel: RSP: 0018:ffffb33b85a7fc98 EFLAGS: 00010296
Apr 21 17:21:26 e69a07245.et15sqa kernel: RAX: 000000000000ff1d RBX: ffff9d356c177d48 RCX: ffff9d356c177d48
Apr 21 17:21:26 e69a07245.et15sqa kernel: RDX: 000000006c171a7d RSI: 0000000000000000 RDI: ffff9d356c177d48
Apr 21 17:21:26 e69a07245.et15sqa kernel: RBP: ffffffffffff9d35 R08: ffffb33b85a7fe18 R09: 0000000000000004
Apr 21 17:21:26 e69a07245.et15sqa kernel: R10: 0000000000010000 R11: ffff9d34658a0000 R12: ffffb33b85a7fe38
Apr 21 17:21:26 e69a07245.et15sqa kernel: R13: 000000006c177d48 R14: ffff9d356c177d48 R15: 00000000000000e3
Apr 21 17:21:26 e69a07245.et15sqa kernel: FS:  00007f3fc0288700(0000) GS:ffff9d357f000000(0000) knlGS:0000000000000000
Apr 21 17:21:26 e69a07245.et15sqa kernel: CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
Apr 21 17:21:26 e69a07245.et15sqa kernel: CR2: 0000559d85f6a250 CR3: 0000003dcda08005 CR4: 00000000003606e0
Apr 21 17:21:26 e69a07245.et15sqa kernel: DR0: 0000000000000000 DR1: 0000000000000000 DR2: 0000000000000000
Apr 21 17:21:26 e69a07245.et15sqa kernel: DR3: 0000000000000000 DR6: 00000000fffe0ff0 DR7: 0000000000000400
Apr 21 17:21:26 e69a07245.et15sqa kernel: Call Trace:
Apr 21 17:21:26 e69a07245.et15sqa kernel:  __tcp_bpf_recvmsg+0x12c/0x350
Apr 21 17:21:26 e69a07245.et15sqa kernel:  tcp_bpf_recvmsg+0x113/0x370
Apr 21 17:21:26 e69a07245.et15sqa kernel:  inet_recvmsg+0x55/0xc0
Apr 21 17:21:26 e69a07245.et15sqa kernel:  __sys_recvfrom+0xc8/0x130
Apr 21 17:21:26 e69a07245.et15sqa kernel:  ? __audit_syscall_entry+0x103/0x130
Apr 21 17:21:26 e69a07245.et15sqa kernel:  ? syscall_trace_enter+0x1d2/0x2d0
Apr 21 17:21:26 e69a07245.et15sqa kernel:  ? __audit_syscall_exit+0x1e4/0x290
Apr 21 17:21:26 e69a07245.et15sqa kernel:  __x64_sys_recvfrom+0x24/0x30
Apr 21 17:21:26 e69a07245.et15sqa kernel:  do_syscall_64+0x4f/0x1b0
Apr 21 17:21:26 e69a07245.et15sqa kernel:  entry_SYSCALL_64_after_hwframe+0x44/0xa9
Apr 21 17:21:26 e69a07245.et15sqa kernel: RIP: 0033:0x7f3fc369182e
Apr 21 17:21:26 e69a07245.et15sqa kernel: Code: d5 53 49 89 f4 89 fb 48 83 ec 10 e8 ec f7 ff ff 45 31 c9 89 c5 45 31 c0 45 89 f2 4c 89 ea 4c 89 e6 89 df b8 2d 00 00 00 0f 05 <48> 3d 00 f0 ff ff 77 36 89 ef 48 89 44 24 08 e8 1e f8 ff ff 48 8b
Apr 21 17:21:26 e69a07245.et15sqa kernel: RSP: 002b:00007f3fc0273630 EFLAGS: 00000246 ORIG_RAX: 000000000000002d
Apr 21 17:21:26 e69a07245.et15sqa kernel: RAX: ffffffffffffffda RBX: 000000000000003c RCX: 00007f3fc369182e
Apr 21 17:21:26 e69a07245.et15sqa kernel: RDX: 0000000000010000 RSI: 00007f3fc0276170 RDI: 000000000000003c
Apr 21 17:21:26 e69a07245.et15sqa kernel: RBP: 0000000000000000 R08: 0000000000000000 R09: 0000000000000000
Apr 21 17:21:26 e69a07245.et15sqa kernel: R10: 0000000000000002 R11: 0000000000000246 R12: 00007f3fc0276170
Apr 21 17:21:26 e69a07245.et15sqa kernel: R13: 0000000000010000 R14: 0000000000000002 R15: 000000000000003c
Apr 21 17:21:26 e69a07245.et15sqa kernel: ---[ end trace 5c5dad4c4a87df7b ]--
```
there are some question：
1、is there sth wrong steps in this job？
2、which kernel you suggest to test cilium + bpf sockops + istio + envoy

**Comment by joestringer** [comment]
This seems like a kernel bug, the kernel should not crash when configuring Cilium. Thanks for the report.

**Comment by seraphsea** [comment]
@joestringer Can you provide a stable running kernel version & cilium version & kube version with sockops in istio + envoy。

**Comment by seraphsea** [comment]
@joestringer run cilium in AWS VM or aliyun VM is OK？or cilium must run in a real host。

**Comment by joestringer** [comment]
I don't personally run sockmap so I don't have a reference version that I can recommend; I believe that it was reasonably stable in Linux v4.19 onwards but there have been some bugs found in the latest versions of Linux. Work is ongoing to fix those.

Your steps look fine, it seems like there was just a regression in one of the newer Linux versions and the CentOS kernel is also affected. CentOS kernels can often pull in newer changes than the version represents (ie CentOS Linux 4.18 is not the same as upstream kernel.org Linux 4.18) so it's hard to say which version began to break sockmap. We won't know without further investigation or a report from investigation of this issue (which is ongoing).

**Comment by jrfastab** [comment]
@seraphsea it _should_ work OK I was not able to reproduce this on the latest development kernels. I'll install the 5.4 kernels and give it a try I guess.

**Comment by jrfastab** [comment]
Found an issue that could cause this. Pushed upstream fix here,
```
https://patchwork.ozlabs.org/project/netdev/patch/158861290407.14306.5327773422227552482.stgit@john-Precision-5820-Tower/
```
when this gets applied it will then be backported into all the fixes branches and we can retest 5.4, etc.
